drivers: eth_mcux power down

agansari opened the pull request:
Power down PHY in eth_mcux driver and set carrier off.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

Depends on zephyrproject-rtos/hal_nxp#24

tbursztyka commented on drivers/ethernet/eth_mcux.c:
```diff
@@ -295,8 +295,11 @@ static void eth_mcux_phy_event(struct eth_context *context)
 			context->phy_state = eth_mcux_phy_state_initial;
+			net_eth_carrier_off(context->iface);
+			ENET_StartSMIWrite(ENET, phy_addr, PHY_BASICCONTROL_REG,
+			   kENET_MiiWriteValidFrame,
```
aling parameters to last '(' + 1 space

tbursztyka commented on west.yml:
```diff
@@ -83,7 +83,7 @@ manifest:
       path: modules/fs/nffs
     - name: hal_nxp
-      revision: c6ce9b7e8ddb501ca6d54e927e1602cb5f7a7183
+      revision: 1d5f331869c32e23c3c7523881aa8f0da95f0792
```
wait that's a different change: you are bumping nxp's HAL to a newer verison. make a dedicated PR for it please.

agansari replied:
@tbursztyka without this version eth_mcux won't work, the version mentioned here has changes related to this patch.

tbursztyka replied:
So make a version bump PR first, and then the eth_mcux PR dependent over it.

agansari replied:
@tbursztyka wouldn't a separate commit work rather than a whole new PR, there is no major SDK update

jukkar replied:
Other option is to add a separate commit into this PR that updates the HAL version.

tbursztyka replied:
the pb with the hal version bump is that it might affect other drivers that are not related to ethernet. Thus why it's imo preferable to separate completely

jukkar replied:
But if we find out that the HAL version does not work everywhere we would be forced to revert both commits anyway so the end result is the same (as if we submit two separate PRs).

agansari replied:
@jukkar should be a good thing, one will not work without the other.

stephanosio replied:
https://docs.zephyrproject.org/latest/guides/modules.html#changes-to-existing-modules

You can use `revision: pull/24/head` to run CI testing. Once it passes, the zephyrproject-rtos/hal_nxp#24 can be merged and the revision here can be updated to the commit hash in the master (i.e. the commit that has been rebased onto master).

agansari committed:
west: hal_nxp update to pull #24

https://github.com/zephyrproject-rtos/hal_nxp/pull/24
Add a mask for KSZ8081 PHY's control register to set power down.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari committed:
drivers: eth_mcux power down

Power down PHY in eth_mcux driver and set carrier off.

Signed-off-by: Andrei Gansari <andrei.gansari@nxp.com>

agansari commented:
@jukkar @tbursztyka i've split west and driver in separate commits and aligned code.

jukkar commented:
Is this PR abandoned?
